core/qt-ble.cpp

```diff
@@ -764,13 +767,47 @@ dc_status_t BLEObject::get_name(char *data, size_t size)
 	return DC_STATUS_SUCCESS;
 }
 
+dc_status_t BLEObject::read_characteristic(const QBluetoothUuid &uuid, char *res, size_t size)
+{
+	for (QLowEnergyService *s: all_services) {
+		const QList<QLowEnergyCharacteristic> chars = s->characteristics();
+		for (const QLowEnergyCharacteristic& ch : chars) {
+			if (ch.uuid() == uuid) {
+				report_info("Reading BLE characteristic %s from service %s",
+					to_str(uuid).c_str(),
+					to_str(s->serviceUuid()).c_str());
+				s->readCharacteristic(ch);
+				QByteArray val = ch.value();
+				if ((size_t)val.size() != size) {
+					report_error("unexpected read characteristic size (%d): expected %lu", val.size(), size);
```

Comment:
I just noted that some of the CICD builds failed - you will need to use `%lld` to print `size`, as some of the compilers used are less tolerant to incorrect format strings.

Reply:
Yeah, I noticed too. I am reading that the platform independent way is to use `"%zu"`

Reply:
But it turns out that the Windows compiler used in CI doesn't like `%zu` so I resorted to use `%lld` as you suggested.